### PR TITLE
Dynamic ServiceMonitor dependency

### PR DESCRIPTION
**Pull request by pepov**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
- Make ServiceMonitor an optional component by creating a controller that adds watches dynamically when the ServiceMonitor CRD gets created
- Replace operator-tools prometheus package with the nested module provided by the upstream prometheus operator

### Why?
The thanos operator is currently not functional without the ServiceMonitor CRD however there could be use cases where it's not needed. This PR makes that possible and also makes standalone testing of the operator easier.

### CODE REVIEW

**Review by bonifaido — APPROVED** [review]

File: controllers/thanos_controller.go [hunk 1]
```diff
@@ -121,11 +121,11 @@ func (r *ThanosReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// QueryFrontend
 	reconcilers = append(reconcilers, query_frontend.New(thanosComponentReconciler).Reconcile)
 
-	return resources.RunReconcilers(reconcilers)
+	result, err := resources.RunReconcilers(reconcilers)
+	return result, err
```
Comment (siliconbrain) [inline controllers/thanos_controller.go:125]: Is this change necessary?

**Review by siliconbrain — APPROVED** [review]
lgtm

File: controllers/thanos_controller.go (hunk 1 above)
Comment (pepov) [reply controllers/thanos_controller.go:125]: no, thanks
